dev/archery/archery/benchmark/core.py

```diff
@@ -0,0 +1,123 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+class Statistics:
+    def __init__(self, min, max, mean, stddev, q1, median, q3):
+        self.min = min
+        self.max = max
+        self.mean = mean
+        self.stddev = stddev
+        self.q1 = q1
+        self.median = median
+        self.q3 = q3
+
+    @staticmethod
+    def from_values(values):
+        sorted(values)
+        n = len(values)
+        mean = sum(values) / len(values)
+        sum_diff = sum([(val - mean)**2 for val in values])
+        stddev = (sum_diff / (n - 1))**0.5 if n > 1 else 0.0
+
+        return Statistics(
+            values[0],
+            values[-1],
+            mean,
+            stddev,
+            values[int(n/4)],
+            values[int(n/2)],
+            values[int((3*n)/4)],
+        )
+
+
+class Benchmark:
+    def __init__(self, name, values, stats=None):
+        self.name = name
+        self.values = values
+        self.statistics = stats if stats else Statistics.from_values(values)
+
+    @property
+    def value(self):
+        return self.statistics.median
+
+    @property
+    def unit(self):
+        return None
+
+    @property
+    def less_is_better(self):
+        return True
+
+
+class BenchmarkSuite:
+    def __init__(self, name, benchmarks):
+        self.name = name
+        self.benchmarks = benchmarks
+
+    def __repr__(self):
+        name = self.name
+        benchmarks = self.benchmarks
+        return f"BenchmarkSuite[name={name}, benchmarks={benchmarks}]"
+
+
+def regress(change, threshold):
```

Comment:
Instead of this, I would probably expect a `Benchmark.does_regress(baseline)` method (that could ultimately take into account the standard deviation and the `less_is_better` property). Of course, that can be later refactored.